### DISCUSSION TITLE
Uploading engines vs Adaptive play

### DISCUSSION THREAD

**Discussion by EdNekebno** [EdNekebno/DGTCentaurMods#78] [Show and tell]
A proper show and tell :) Today I worked on this...I'll include it when I next push. Basically it allows you to upload pre-compiled engine files and a .uci file which specifies some settings (there can be various options). I modified the menu so if the .uci file exists then it will show you the various options for play included in it. If not then it will just play straight. 

By doing it this way somebody can uploaded any uci engine that's compiled for the zero w and play it. Which is pretty cool. It keeps this repository and people's menu screens tidy too as we don't need to pre-include them all (although I might set up a different repository with different engines and .uci files).

![engines](https://user-images.githubusercontent.com/82707095/140621067-78f1c947-b41c-41e5-b47f-801e8c27c88e.png)

**Comment by wormstein** [comment]
From what I read and see, this is a game changer!!! Great work Ed!

You could set up a new repo or just a file in DGTM where we could maintain a list of engines and repo locations (maybe a db table better) that we offer to the user. Once they select it, download and compilation begins. Course leave this option too. I am afraid some will just don't know about architecture :)))

PS: doctors said Monday sounds promising. Still in hosp, recovering...

**Reply by EdNekebno** [reply]
I actually started on an adaptive script to replace the centaur process if people didn't want to install the centaur stuff - but then ditched it for this. Adaptive mode has problems that the computer is always going to have a good opening and endgame and the midgame will be very close. Then it's essentially random if you win in the endgame. That's actually my experience with the centaur software - so I realised I like playing engines more as they're more realistic/variable (I don't use centaur mode at all nowadays) and concentrated on that.

I really like the idea of some sort of lookup and install system like you describe. I'm envisioning that most people will have a handful of their favorite engines installed but might want to try a new one from time to time. It'd be like an app store for chess engines :)

Rest and recover...hope it's not too boring.

**Reply by wormstein** [reply]
An app store for the board doesn't sound bad at all!!!

**Reply by cyberluke** [reply]
Interesting note about adaptive mode. So in the end, it could work only if you have player profiles and track a history of ELO. Then you can adapt to that based on a previous game, not previous turn.
